### DISCUSSION TITLE
Redirect /gems/magiclink to /gems/magic-link

### DISCUSSION THREAD

**Discussion by bradgessler** [rubygems/rubygems.org#2976] [Ideas]
I recently went to publish a gem that I thought was free and available, but got this error:

```
$ magiclink git:(main) rake release
magiclink 0.1.0 built to pkg/magiclink-0.1.0.gem.
Tag v0.1.0 has already been created.
Pushing gem to https://rubygems.org...
There was a problem saving your gem: Name 'magiclink' is too similar to an existing gem named 'magic-link'
```

When I view that webpage, it looks like its fair game:

<img width="1381" alt="Screen Shot 2022-02-28 at 16 00 13" src="https://user-images.githubusercontent.com/4628/156079156-89580088-987b-4c74-80c0-a58bb2a18ebe.png">

I spent 20-30 min renaming the gem from something else to `magiclink`. That's fine, but what pains me is to think about the number of other people that will try this for 20-30 min, push, and get rejected.

To prevent this from happening (and saving lots of dev time) I propose that urls like `https://rubygems.org/gems/magiclink` get redirects to `https://rubygems.org/gems/magic-link` or some message is displayed informing the dev that they can't have that gem name.

**Comment by simi** [comment]
Sounds good, we do already render something special for blacklisted gems here.

https://github.com/rubygems/rubygems.org/blob/47ddef40268ea66ecc2bbdf5c0ddb996cd5784d8/app/controllers/rubygems_controller.rb#L22-L25

Gem typo is checked here in pusher.

https://github.com/rubygems/rubygems.org/blob/47ddef40268ea66ecc2bbdf5c0ddb996cd5784d8/app/models/rubygem.rb#L372-L377

It should be possible to update GemTypo model to return which already existing `Rubygem` is in conflict and redirect in there with flash message explaining what has happened.

Would you mind to contribute?

**Reply by bradgessler** [reply]
Sure! I'll give it a go if you're the person who would be able to review and merge the PR if the implementation looks good. I don't want to spend time hacking on it and have another reviewer reject it on principal.

**Reply by simi** [reply]
Yup, I can help you to get merged this. Ping me at PR description.
